packages/webapp/src/components/Task/AddSoilAmendmentProducts/QuantityApplicationRate/useQuantityApplicationRate.ts

```diff
@@ -139,33 +144,37 @@ export const useQuantityApplicationRate = ({
   }, [application_rate_weight_unit, application_rate_volume_unit]);
 
   /* For the preview string, replicate the conversion the unit component would do if the value had been displayed in a <Unit /> rather than a string. However, don't update upon changes to application_area_unit beyond the initial undefined > defined change */
-  const [previewStringValue, setPreviewStringValue] = useState<number | null>(null);
-  const [previewStringUnit, setPreviewStringUnit] = useState<string | null>(null);
+  const previewStrings = useMemo(() => {
```

Comment:
Just a question about how you are using this to "freeze" the value. I don't think this needs to be optimized.

React docs:

> You should only rely on useMemo as a performance optimization. If your code doesn’t work without it, find the underlying problem and fix it first. Then you may add useMemo to improve performance.

Reply:
Wait, I also found a new issue with this now that the percent value is retained even when switching locations so I'm pretty sure gonna change the code again! 😓 But to clarify are you asking a) how does the useMemo function to freeze the value, or b) why is it being used to freeze the value when it's intended for optimization/

Reply:
Mostly b) !